src/main/java/htsjdk/samtools/cram/structure/Slice.java

```diff
@@ -337,51 +372,42 @@ public CramRecordReader createCramRecordReader(final CompressionHeader header,
         return reader.getReferenceSpans();
     }
 
-    /**
-     * Generate a CRAI Index entry from this Slice
-     * @return a new CRAI Index Entry derived from this Slice
-     */
-    public CRAIEntry getCRAIEntry() {
-        return new CRAIEntry(referenceContext.getSerializableId(), alignmentStart, alignmentSpan, containerOffset, offset, size);
-    }
-
     /**
      * Generate a CRAI Index entry from this Slice and other container parameters,
      * splitting Multiple Reference slices into constituent reference sequence entries.
      *
-     * TODO: investigate why we sometimes need to pass in an external containerStartByteOffset
-     * because this Slice's containerOffset is incorrect
-     *
-     * TODO: investigate why we sometimes split multi-ref and sometimes do not
-     *
      * @param compressionHeader the enclosing {@link Container}'s CompressionHeader
-     * @param landmarks the enclosing {@link Container}'s landmarks
-     * @param containerStartByteOffset the byte offset of the enclosing {@link Container}
      * @return a list of CRAI Index Entries derived from this Slice
      */
-    public List<CRAIEntry> getCRAIEntriesSplittingMultiRef(final CompressionHeader compressionHeader,
-                                                           final int[] landmarks,
-                                                           final long containerStartByteOffset) {
+    public List<CRAIEntry> getCRAIEntries(final CompressionHeader compressionHeader) {
         if (! compressionHeader.isCoordinateSorted()) {
             throw new CRAMException("Cannot construct index if the CRAM is not Coordinate Sorted");
         }
 
+        craiIndexInitializationCheck();
+
         if (referenceContext.isMultiRef()) {
             final Map<ReferenceContext, AlignmentSpan> spans = getMultiRefAlignmentSpans(compressionHeader, ValidationStringency.DEFAULT_STRINGENCY);
 
             return spans.entrySet().stream()
                     .map(e -> new CRAIEntry(e.getKey().getSerializableId(),
                             e.getValue().getStart(),
                             e.getValue().getSpan(),
-                            containerStartByteOffset,
-                            landmarks[index],
-                            size))
+                            containerByteOffset,
+                            byteOffsetFromContainer,
+                            byteSize))
                     .sorted()
                     .collect(Collectors.toList());
         } else {
             // single ref or unmapped
             final int sequenceId = referenceContext.getSerializableId();
-            return Collections.singletonList(new CRAIEntry(sequenceId, alignmentStart, alignmentSpan, containerStartByteOffset, offset, size));
+            return Collections.singletonList(new CRAIEntry(
+                    sequenceId,
+                    alignmentStart,
+                    alignmentSpan,
+                    containerByteOffset,
```

Comment:
`containerByteOffset` newly uses the Slice member variable.
`byteOffsetFromContainer` and `byteSize` are continuing uses of the Slice member variables, renamed.